Tests/HarmonyTests/PutNetworkDataSourceTests.swift

```diff
@@ -0,0 +1,394 @@
+        let expectation = XCTestExpectation(description: "expectation")
+
+        dataSource.put(value, in: query)
+            .then { value in
+                if expectedError == nil {
+                    expectation.fulfill()
+                }
+            }
+            .fail { error in
+                if let expectedError = expectedError {
+                    if type(of: error) == type(of: expectedError) {
+                        expectation.fulfill()
+                    }
+                }
+            }
+
+        wait(for: [expectation], timeout: 1.0)
+    }
+
+    private func providePutDataSource<S: Decodable>(
+        url: String,
+        request: URLRequest? = nil,
+        response: URLResponse? = nil,
+        decoder: JSONDecoder? = nil,
+        jsonFileName: String? = nil
+    ) -> PutNetworkDataSource<S> {
+        let session = Utils.provideMockAlamofireSession(request: request, response: response, jsonFileName: jsonFileName)
+        return PutNetworkDataSource<S>(url: url, session: session, decoder: decoder ?? DecoderSpy())
+    }
+
+    private func expectPutError<S: Decodable>(
+        value: S? = nil,
+        _ dataSource: PutNetworkDataSource<S>,
+        _ query: Query,
+        _ expectedError: Error?,
+        _ function: Function)
+    {
+        if function == .putAll {
+            expectPutAll(value: (value != nil) ? [value!] : [], dataSource, query, expectedError)
+        } else {
+            expectPut(value: value, dataSource, query, expectedError)
+        }
+    }
+    
+    private func expectPutAllError<S: Decodable>(
+        value: [S],
+        _ dataSource: PutNetworkDataSource<S>,
+        _ query: Query,
+        _ expectedError: Error?,
+        _ function: Function)
+    {
+        expectPutAll(value: value, dataSource, query, expectedError)
+    }
+    
+    private func expectPutAlamofireError<S: Decodable>(
+            _ dataSource: PutNetworkDataSource<S>,
+            _ query: Query,
+            _ expectedError: AFError)
+    {
+        let expectation = XCTestExpectation(description: "expectation")
+
+        dataSource.put(nil, in: query).then { _ in
+            print("")
+        }.fail { error in
+            if let error = error as? AFError {
+                if error.localizedDescription == expectedError.localizedDescription {
+                    expectation.fulfill()
+                }
+            }
+        }
+
+        wait(for: [expectation], timeout: 1.0)
+    }
+    
+    private func expectPutAllAlamofireError<S: Decodable>(
+            value: [S],
+            _ dataSource: PutNetworkDataSource<S>,
+            _ query: Query,
+            _ expectedError: AFError)
+    {
+        let expectation = XCTestExpectation(description: "expectation")
+
+        dataSource.putAll(value, in: query).then { _ in
+            print("")
```

Comment:
Remove it, you could just remove the `them{}` function directly, as it's not needed.

Reply:
done